src/components/partnership/partnership.repository.ts

```diff
@@ -195,25 +222,24 @@ export class PartnershipRepository extends DtoRepository<
               : q,
           ),
       )
-
       .match(requestingUser(session))
       .apply(
         this.privileges.forUser(session).filterToReadable({
           wrapContext: oncePerProject,
         }),
       )
       .apply(sorting(Partnership, input))
-      .apply(paginate(input))
+      .apply(paginate(input, this.hydrate(session)))
```

Comment:
`hydrate` has not been overridden here to provide the correct shape.
Notice how `readMany`, or anything, doesn't call it either.

Tracing this was simple:
```
TypeError: Cannot read properties of undefined (reading 'id')

    at SyncBudgetRecordsToFundingPartners.syncRecords (src/components/budget/handlers/sync-budget-records-to-funding-partners.handler.ts:152:53)at SyncBudgetRecordsToFundingPartners.handle (src/components/budget/handlers/sync-budget-records-to-funding-partners.handler.ts:94:18)
    at SyncEventBus.publish (src/core/events/event-bus.service.ts:48:7)
    at ProjectService.update (src/components/project/project.service.ts:316:5)
    at Connection.withTransaction [as runInTransaction] (src/core/database/transaction.ts:96:12)
    at ProjectService.descriptor.value (src/core/database/transactional.decorator.ts:44:14)
    at ProjectResolver.updateProject (src/components/project/project.resolver.ts:374:21)
    at Object.<anonymous> (test/project-changeset-aware.e2e-spec.ts:211:5)
```
1. `syncRecords` is called from the list result of [determinePartnerships](https://github.com/SeedCompany/cord-api-v3/blob/0937-edgedb-partnership-queries/src/components/budget/handlers/sync-budget-records-to-funding-partners.handler.ts#L91-L91)
2. `determinePartnerships` has several condition branches. We know it's the [`ProjectUpdateEvent`](https://github.com/SeedCompany/cord-api-v3/blob/0937-edgedb-partnership-queries/src/components/budget/handlers/sync-budget-records-to-funding-partners.handler.ts#L137-L138), because `ProjectService.update` is what is in the stacktrace.
3. So I put a breakpoint on the [result of the partnerships list](https://github.com/SeedCompany/cord-api-v3/blob/0937-edgedb-partnership-queries/src/components/partnership/partnership.repository.ts#L234-L234)
4. Ran the test that failed, `project-changeset-aware.e2e-spec.ts:211`, given by stacktrace.
5. Saw the was missing all of the links/relationships. Why? Because the [base hydrate](https://github.com/SeedCompany/cord-api-v3/blob/0937-edgedb-partnership-queries/src/core/database/dto.repository.ts#L153-L154) only grabs the direct props.

A go-to on `hydrate` would've revealed this as well.
Or a `.logIt()` on the query showing that none of the relationships were matched/merged.

Reply:
I walked through this with Will and we track with you the whole way down to 5).  At that point I'm confused though.  From your red/green comment above it looks like you want him to add in the `this.hydrate(session)`, but that already exists on that line, so that doesn't make sense.

From your explanation, on the other hand, I think you're saying that we need to add a `hydrate` method specifically inside the Neo4j repo that actually returns the relationships as well, instead of just falling back on the base `DtoRepository` hydrate method.

Can you clarify for us here?

Reply:
> From your explanation, on the other hand, I think you're saying that we need to add a `hydrate` method specifically inside the Neo4j repo that actually returns the relationships as well, instead of just falling back on the base `DtoRepository` hydrate method.

That is correct!



> From your red/green comment above it looks like you want him to add in the `this.hydrate(session)` portion of that line, but that already exists, so that doesn't make sense.

You guys added the first call to `hydrate` here, not me. And the default implementation was insufficient to complete the shape we require. But we have no way of confirming that, it's just typecasts with cypher/neo4j.